push-notifications/android/src/main/java/com/capacitorjs/plugins/pushnotifications/PushNotificationsPlugin.java

```diff
@@ -128,13 +129,13 @@ public void getDeliveredNotifications(PluginCall call) {
     public void removeDeliveredNotifications(PluginCall call) {
         JSArray notifications = call.getArray("notifications");
 
-        List<Integer> ids = new ArrayList<>();
+        List<String> tags = new ArrayList<>();
         try {
             for (Object o : notifications.toList()) {
                 if (o instanceof JSONObject) {
                     JSObject notif = JSObject.fromJSONObject((JSONObject) o);
-                    Integer id = notif.getInteger("id");
-                    ids.add(id);
+                    String tag = notif.getString("tag");
```

Comment:
RemoveDeliveredNotifications removes both push and local notifications, local notifications don't have a tag (at the moment, could be added in the future).
And on local notifications the id is not 0.
So the code should check if the tag is not null and call the `cancel(tag, id)`, and if null call `cancel(id)`.
For making things easier, you can remove the tags/ids list and the for where cancel is called and just call cancel here instead of the `tags.add(tag);`